src/overlays/actors/ovl_Door_Warp1/z_door_warp1.c

```diff
@@ -70,19 +70,27 @@ void DoorWarp1_SetupAction(DoorWarp1* this, DoorWarp1ActionFunc actionFunc) {
     this->actionFunc = actionFunc;
 }
 
-s32 func_808B849C(DoorWarp1* this, PlayState* play) {
-    s32 ret = 0;
+/**
+ * Returns the number of which remains are obtained inside this warp.
+ * * 0: None (if it's not a boss or the remains were already obtained)
+ * * 1: Odolwa
+ * * 2: Goht
+ * * 3: Gyorg
+ * * 4: Twinmold
+ */
+s32 DoorWarp1_GetRemains(DoorWarp1* this, PlayState* play) {
+    s32 remains = 0;
 
     if ((play->sceneId == SCENE_MITURIN_BS) && !CHECK_QUEST_ITEM(QUEST_REMAINS_ODOLWA)) {
-        ret = 1;
+        remains = 1;
     } else if ((play->sceneId == SCENE_HAKUGIN_BS) && !CHECK_QUEST_ITEM(QUEST_REMAINS_GOHT)) {
-        ret = 2;
+        remains = 2;
```

Comment:
Github doesn't let me to add suggestions to all these cases at once, so I'm adding them individually.

```suggestion
        remains = 1 + GI_REMAINS_GOHT - GI_REMAINS_ODOLWA;
```